### PR TITLE
OCPBUGS-15599: configurable ironic agent vlan creation

### PR DESCRIPTION
**Pull request by tdomnesc**
Disable vlan creation by default if nmstate is provided
  
 IRONIC_AGENT_VLAN_INTERFACES env variable can be used to force "all" or "" in ironic agent config regardless of nmstate
 
 https://issues.redhat.com/browse/OCPBUGS-15599

### CODE REVIEW

**Comment by dtantsur** [comment]
@tdomnesc could you add unit tests for the default behavior to check that it depends on nmstate?

**Comment by tdomnesc** [comment]
@dtantsur - if nmstate is not empty we run an external command: `exec.Command("nmstatectl", "gc", "/dev/stdin")`. I thought about it and can't come up with any decent way to unit test this.

File: pkg/ignition/service_config_test.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func TestIronicPythonAgentConf(t *testing.T) {
 				ironicBaseURL:          tt.ironicBaseURL,
 				ironicInspectorBaseURL: tt.ironicInspectorBaseURL,
 			}
-			if got := b.IronicAgentConf(); !reflect.DeepEqual(got, tt.want) {
+			if got := b.IronicAgentConf("all"); !reflect.DeepEqual(got, tt.want) {
```
Comment (dtantsur) [inline pkg/ignition/service_config_test.go:41]: Please add another test that uses a different value than "all".

**Pull request merged**